src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/EnumGroupAndName.cs

```diff
@@ -28,7 +30,23 @@ public EnumGroupAndName(string group, string name)
             }
 
             Group = group;
-            Name = name;
+            _nameFunc = () => name;
+        }
+
+        public EnumGroupAndName(string group, Func<string> name)
```

Comment:
Yeah I'm not sure if it needs to be obsoleted. If the docs just said precisely what it did and didn't do, that would seem to be fine.